components/chart/editModule/index.tsx

```diff
@@ -0,0 +1,282 @@
+
+  const { control, handleSubmit, formState: { errors }, register, setValue, getValues } = useForm<any>();
+  useEffect(() => {
+    if (targetModel && data?.contentInstances.length > 0) {
+      targetModel?.ViewFields.reduce(
+        (obj, field) => {
+          const fieldData = data.contentInstances[0].FieldsValues.find(
+            (f) => (f.FieldId === field.Id),
+          );
+          if (field.Type === ModelFieldsTypes.PredefinedListCheckboxes) {
+            const fieldDataForCheckboex = data.contentInstances[0].FieldsValues.filter(
+              (f) => f.FieldId === field.Id,
+            );
+            const finishedCheckedBox = fieldDataForCheckboex.map((checkBoxValue) => checkBoxValue.Value.split(','))
+            setValue(`${field.Id}`, finishedCheckedBox.flat() || '')
+          } else if (field.Type === ModelFieldsTypes.Boolean) {
+            data.contentInstances[0].FieldsValues.filter(
+              (f) => f.FieldId === field.Id,
+            ).map((item) => {
+              setValue(`${field.Id}`, item.Value === 'True' ? true : false)
+            });
+          } else {
+            setValue(`${field.Id}`, fieldData?.Value || '')
+          }
+
+          return {
+            ...obj,
+            [`${field.Id}`]: fieldData?.Value || '',
+          };
+        },
+        {}
+      );
+    }
+  }, [data, targetModel]);
+
+  const onSubmit = (formData: FieldValues) => {
+    const updatedFormData: FieldValues = { ...formData };
+    let ci: number;
+    if (targetModel && data?.contentInstances.length > 0) {
+      targetModel?.ViewFields.reduce(
+        (obj, field) => {
+          if (field.Type === ModelFieldsTypes.PredefinedListCheckboxes) {
+            updatedFormData[field.Id] = formData[field.Id].join(',')
+          } else if (field.Type === ModelFieldsTypes.Boolean) {
+            updatedFormData[field.Id] = formData[field.Id] === true ? 'True' : 'False'
+          }
+          data.contentInstances[0].FieldsValues.map((details, index) => {
+            ci = details.ContentId;
+          });
+          return {
+            ...obj,
+          };
+        },
+        {}
+      );
+    }
+
+    updateContentInstanceFieldValues({
+      variables: {
+        body: {
+          Id: ci,
+          FieldsValues: Object.keys(formData).map((key) => ({
+            FieldId: parseInt(key.replace(ci + '-', '')),
+            Value: updatedFormData[key],
+            ContentId: ci,
+          })),
+        },
+      },
+    })
+      .then(() => {
+        setSuccessMessage('Content instance updated successfully');
+        setErrorMessage('');
+        setTimeout(() => {
+          setSuccessMessage('');
+        }, 3000);
+      })
+      .catch((error) => {
+        setErrorMessage(`${error}: Error updating content instance`);
+        setSuccessMessage('');
+        setTimeout(() => {
+          setSuccessMessage('');
+        }, 3000);
+      });
+
+  };
+  const { enumValues } = useTagsEnumValuesQuery(403027);
```

Comment:
let's move this one with other queries, and let's add maybe 403027 as a constant in Constants.ts